pkg/aggregator/demultiplexer_mock.go

```diff
@@ -52,40 +52,43 @@ func (a *TestAgentDemultiplexer) AddLateMetrics(metrics metrics.MetricSampleBatc
 	a.Unlock()
 }
 
-func (a *TestAgentDemultiplexer) samples() []metrics.MetricSample {
+func (a *TestAgentDemultiplexer) samples() ([]metrics.MetricSample, []metrics.MetricSample) {
```

Comment:
All looks good, just a small suggestion (mainly as I'll forget which is which)
```suggestion
func (a *TestAgentDemultiplexer) samples() (received []metrics.MetricSample, late []metrics.MetricSample) {
```
Same for `WaitForSamples` on line 72.

Reply:
Or should that `(received []metrics.MetricSample, timed []metrics.MetricSample)`?

Reply:
Good catch, I'll used `timed`.

Note that after a discussion with Maxime I also have a ticket around homogenizing this for TimedMetric or TimedSamples in the API and such, but it is opening a few questions I have to solve (TimedSample is way similar-looking to TimeSampler).